solr/core/src/java/org/apache/solr/update/UpdateHandler.java

```diff
@@ -115,12 +115,12 @@ public UpdateHandler(SolrCore core, UpdateLog updateLog)  {
     boolean skipUpdateLog = core.getCoreDescriptor().getCloudDescriptor() != null && !core.getCoreDescriptor().getCloudDescriptor().requiresTransactionLog();
     if (updateLog == null && ulogPluginInfo != null && ulogPluginInfo.isEnabled() && !skipUpdateLog) {
       DirectoryFactory dirFactory = core.getDirectoryFactory();
-      if (dirFactory instanceof HdfsDirectoryFactory) {
-        ulog = new HdfsUpdateLog(((HdfsDirectoryFactory)dirFactory).getConfDir());
-      } else {
-        ulog = ulogPluginInfo.className == null ? new UpdateLog():
-                core.getResourceLoader().newInstance(ulogPluginInfo, UpdateLog.class, true);
-      }
+
+      // if the update log class is not defined in the plugin info / solrconfig.xml
+      // (like <updateLog class="${solr.ulog:solr.UpdateLog}"> )
+      // we fall back use the one which is the default for the given directory factory
+      ulog = ulogPluginInfo.className == null ? dirFactory.newDefaultUpdateLog() :
```

Comment:
but then uLogPluginInfo would be ignored, which might have settings to be applied.  Either `newDefaultUpdateLog` should take the pluginInfo as a parameter, or just have it return the class (or class name) to be instantiated (and maybe tweak the name to be getDefaultUpdateLogClassName).  I think the latter is cleaner.

Reply:
@dsmiley  not sure I understand your comment here. The original logic:

```
DirectoryFactory dirFactory = core.getDirectoryFactory();
      if (dirFactory instanceof HdfsDirectoryFactory) {
        ulog = new HdfsUpdateLog(((HdfsDirectoryFactory)dirFactory).getConfDir());
      } else {
        ulog = ulogPluginInfo.className == null ? new UpdateLog():
                core.getResourceLoader().newInstance(ulogPluginInfo, UpdateLog.class, true);
      }
```

If `ulogPluginInfo.className == null` then `new UpdateLog();`. So I don't think there is any change here to the logic for `UpdateLog()`. Line 132 has `ulog.init(ulogPluginInfo);` which I think makes sure the ulog implementation takes into account the `uLogPluginInfo).

Reply:
Ooh; I missed line 132.  Thanks.